scipy/sparse/csgraph/_shortest_path.pyx

```diff
@@ -515,67 +535,198 @@ def dijkstra(csgraph, directed=True, indices=None,
 
     #------------------------------
     # initialize dist_matrix for output
-    dist_matrix = np.zeros((len(indices), N), dtype=DTYPE)
-    dist_matrix.fill(np.inf)
-    dist_matrix[np.arange(len(indices)), indices] = 0
+    if min_only:
+        dist_matrix = np.zeros((N), dtype=DTYPE)
```

Comment:
`dist_matrix = np.full(N, np.inf, dtype=DTYPE)` would be simpler here, and in the other cases below.